hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -748,6 +753,26 @@
     public static final HazelcastProperty QUERY_OPTIMIZER_TYPE
             = new HazelcastProperty("hazelcast.query.optimizer.type", QueryOptimizerFactory.Type.RULES.toString());
 
+    /**
+     * Type of Query Index result copying behavior.
+     * Valid Values:
+     * <ul>
+     * <li>READ - Internal data structures of the index are concurrently modified without copy-on-write semantics.
+     * Index queries copy the results of a query on index read to detach the result from the source map.
+     * Should be used in index-write intensive cases. (default value).</li>
+     * <li>WRITE - Internal data structures of the index are modified with copy-on-write semantics.
+     * Previously returned indexed query results reflect the state of the index at the time of the query and are not
+     * affected by future index modifications.</li></li>
+     * <li>NEVER - Internal data structures of the index are concurrently modified without copy-on-write semantics.
+     * Index reads never copy the results of a query to a separate map.
+     * It means that the results backed by the underlying index-map can change after the query has been executed.
+     * Specifically an entry might have been added / removed from an index, or it might have been remapped.</li>
+     * </ul>
+     * <p/>
+     */
+    public static final HazelcastProperty QUERY_INDEX_RESULT_COPY
```

Comment:
To be consistent with the code this property can also be called as 'INDEX_BOPY_BEHAVIOUR`?

Reply:
👍